### DISCUSSION TITLE
Unable to get example to work

### DISCUSSION THREAD

**Discussion by MastaBaba** [tectalichq/public-openai-client-php#36] [Q&A] [answered]
I'm trying to get the first example in the docs to work, but I'm failing.  Here's my full code:

```
	ini_set('display_errors', '1');
	ini_set('display_startup_errors', '1');
	error_reporting(E_ALL);

	// Load your project's composer autoloader (if you aren't already doing so).
	require_once(__DIR__ . '/vendor/autoload.php');
	
	use Symfony\Component\HttpClient\Psr18Client;
	use Tectalic\OpenAi\Authentication;
	use Tectalic\OpenAi\Client;
	use Tectalic\OpenAi\Manager;

	$api_key = "{{key}}";
	
	$openaiClient = \Tectalic\OpenAi\Manager::build(
		new \GuzzleHttp\Client(),
		new \Tectalic\OpenAi\Authentication(getenv($api_key))
	);
	
	/** @var \Tectalic\OpenAi\Models\ChatCompletions\CreateResponse $response */
	$response = $openaiClient->chatCompletions()->create(
		new \Tectalic\OpenAi\Models\ChatCompletions\CreateRequest([
			'model' => 'gpt-4',
			'messages' => [
				[
					'role' => 'user',
					'content' => 'Will using a well designed and supported third party package save time?'
				],
			],
		])
	)->toModel();
	
	echo $response->choices[0]->message->content;
```

This is what I get:

> Fatal error: Uncaught Tectalic\OpenAi\ClientException: Unsuccessful response. HTTP status code: 401 (Unauthorized). in {{folder}}/openai/vendor/tectalic/openai/src/Handlers/ChatCompletions.php:150 Stack trace: #0 {{folder}}/openai/index.php(36): Tectalic\OpenAi\Handlers\ChatCompletions->toModel() #1 {main} thrown in {{folder}}/openai/vendor/tectalic/openai/src/Handlers/ChatCompletions.php on line 150

It is my understanding that the variable $openaiClient is called with authentication. So, where am I going wrong?

I use an API key that I created an hour before running this code.

**Comment by thejamescollins** [comment, marked as answer]
Hi @MastaBaba,

I think these two lines may be the issue:

```php
$api_key = "{{key}}";
```

```php
getenv($api_key)
```

Are you storing your API key in an environment variable? The [getenv() PHP function](https://www.php.net/manual/en/function.getenv.php) loads the value from the environment variable named`{{key}}`.

I'd encourage you to use an environment variable for your API key.

Please see the docs for more details: https://tectalic.com/apis/openai/docs/php#usage

Or if you aren't using an environment variable and you are hard-coding it in the PHP code itself, then you would need to remove the `getenv()` call.

I hope that helps,

James

**Comment by MastaBaba** [comment]
Thanks @om4james. Indeed, I was not familiar with getting environment variables in PHP, so misunderstood this function.

I got it working now.

**Comment by thejamescollins** [comment]
That's great news @MastaBabal, thank you for letting me know.

All the best with your project.

James
